src/ZipkinBundle/Components/HttpClient/DefaultHttpParser.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace ZipkinBundle\Components\HttpClient;
+
+use Zipkin\SpanCustomizer;
+use const Zipkin\Tags\HTTP_PATH;
+use const Zipkin\Tags\HTTP_METHOD;
+use const Zipkin\Tags\HTTP_STATUS_CODE;
+use const Zipkin\Tags\ERROR;
+use Symfony\Contracts\HttpClient\ResponseInterface;
+
+class DefaultHttpParser implements HttpParser
+{
+    public function request(string $method, string $url, array $options = [], SpanCustomizer $span)
```

Comment:
Consider RequestInterface ? because when this pulls up from symphony it allows a base parser for different HTTP libraries. Also adding new things, such as length or authority is easier

Reply:
Unfortunately Symfony chose to use this signature for doing a request (notice `$options` hold a lot of information about the request https://github.com/symfony/http-client-contracts/blob/master/HttpClientInterface.php#L26):

```php
public function request(string $method, string $url, array $options = []): ResponseInterface
```

So this `HttpParser` is ad-hoc for symfony. In zipkin-php we are going to add one using the standard PHP `RequestInterface`.

Reply:
just wondering why in our parser we can't use RequestInterface anyway..

Reply:
We could have our RequestInterface but that means I create an additional object and in default case I will access the information I already have which is not too problematic but since this is Symfony and they usually _swim against the current_ probably better to keep their signature.